packages/reactive-element/src/legacy-decorators/query-all.ts

```diff
@@ -38,14 +62,26 @@ import {decorateProperty} from './base.js';
  * ```
  * @category Decorator
  */
-export function queryAll(selector: string) {
-  return decorateProperty({
-    descriptor: (_name: PropertyKey) => ({
-      get(this: ReactiveElement) {
-        return this.renderRoot?.querySelectorAll(selector) ?? [];
-      },
-      enumerable: true,
-      configurable: true,
-    }),
-  });
+export function queryAll(selector: string): QueryAllDecorator {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (<C extends ReactiveElement, V extends NodeList>(
+    protoOrTarget: ClassAccessorDecoratorTarget<C, V>,
+    nameOrContext: PropertyKey | ClassAccessorDecoratorContext<C, V>
+  ) => {
+    if (typeof nameOrContext === 'object') {
+      return standardQueryAll(selector)(
+        protoOrTarget,
+        nameOrContext as ClassAccessorDecoratorContext<C, V>
+      );
+    } else {
+      Object.defineProperty(protoOrTarget, nameOrContext as PropertyKey, {
+        get(this: ReactiveElement) {
+          return this.renderRoot?.querySelectorAll(selector) ?? [];
+        },
+        enumerable: true,
+        configurable: true,
+      });
+      return;
```

Comment:
Is an empty explicit return required? Can this be omitted.

Reply:
It's required by [noImplicitReturns](https://www.typescriptlang.org/it/tsconfig#noImplicitReturns)